### ISSUE TITLE
You can't shoot over quarterpannels

### ISSUE THREAD

**Issue by Salty-Panda** [CleverRaven/Cataclysm-DDA#40249]
# Describe the bug

I added a shooting range to my car in order to train skills/fight safely.

I tried various ranged weapons: guns, bow, throwing, flamethrowers.
None of those weapons auto-targeted enemies after using the _f_ button. 
After manual targeting, only arrows were able to leave vehicle's borders, properly hit and damage enemies.
Grenades and napalm landed at my feet/at quarterpannel in front of me. 
In case of guns, empty casings dropped around me normally, but I didn't hear any sound suggesting that bullets are colliding with the vehicle neither any zombie got damaged.

# Steps To Reproduce

Vehicle's construction as numbered in the screenshot:

1. Military composite ram
2. Military composite ram
3. Military composite ram
4. Orichalcum frame, steel plating, heavy duty quarterpannel
5. Orichalcum frame, steel plating, heavy duty quarterpannel
6. Orichalcum frame, steel plating, heavy duty quarterpannel, inboard mirror
7. Orichalcum frame, steel plating, reinforced headlight, door motor, curtain, reinforced windshield
8. Orichalcum frame, steel plating, reinforced headlight, door motor, aisle curtain, heavy duty hatch, security camera
9. Orichalcum frame, steel plating, reinforced headlight, heavy duty board

# Expected behavior

Being able to use all kinds of weapons over quarterpannels, not only bows and vehicle turrets.

# Screenshots

![image](https://user-images.githubusercontent.com/26396019/81158437-438f2400-8f88-11ea-9293-fb33dea47ec2.png)

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.E-1231-g4968f19 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    No Fungal Monsters [No_Fungi],
    Magiclysm [magiclysm],
    Aftershock [aftershock],
    DinoMod [DinoMod],
    Alternative Map Key [alt_map_key],
    Mining Mod [Mining_Mod],
    No Rail Stations [No_Rail_Stations],
    Graphical Overmap [Graphical_Overmap]
]

**Comment by cirops** [comment]
(Bear with me, I'm just dipping my toes into debugging this game, and maybe quicken the problem-solving by the real devs).
It seems to me that any vehicle part with the flag OBSTACLE will neither let you walk through, nor shoot through. As soon as you remove that flag from the quarterpanels, you can both walk and shoot through them.

So I suppose the solutions could be:
1. Remove the OBSTACLE flag from the quarterpanels, which would be kind of realistic (as in, you can actually "hop" through quarterpanels), but a major buff to monsters, as they would be able to do that too).
2. Add another flag, like SHOOT_THROUGH, to quarterpanels, which the targetting system would have to be modified to recognize, so that you wouldn't be able to walk, but would be able to shoot-through.
3. Keep the system as-is and not let the player shoot through quarterpanels at all.

Hope I helped a bit.

**Comment by LeahLuong** [comment]
This is especially annoying when enemies are @ a slight diagonal to an open door in a vehicle in which you're sitting. Say, 7 tiles left of the doorway & 1 up, the bullet's path will always include the quarterpanel up & left of your char. This is countertuitive from both realism & gameplay considerations: in reality, 1 would simply lean slightly w/out shifting position in order to get a clear shot; drawing a line from the center of your tile to the center of the enemy's tile should only include the quarterpanel's tile when the enemy's <3 tiles left of your char. If they're in the diagonal sweet spot (for them), shifting position (say, down) in the vehicle still results in your shot getting blocked by a quarterpanel.

If the above suggestion for a "SHOOT_THROUGH" flag can't be implemented, perhaps something akin to the cover % for terrain/furniture could be. Vehicle boards block vision so it makes sense they'd block gunfire but the reverse logic seemingly doesn't apply to quarterpanels. What about ignoring cover %'s for adjacent tiles when it comes to shooting outwards??

**Comment by Fris0uman** [comment]
duplicate of #22541
